php-nginx/composer.sh

```diff
@@ -49,6 +49,36 @@ EOF
         ln -sf ${PHP56_DIR} ${PHP_DIR}
     fi
 
+    # Handle custom oauth keys (Adapted from https://github.com/heroku/heroku-buildpack-php/blob/master/bin/compile)
+    COMPOSER_GITHUB_OAUTH_TOKEN=${COMPOSER_GITHUB_OAUTH_TOKEN:-}
+    if [[ -n "$COMPOSER_GITHUB_OAUTH_TOKEN" ]]; then
+        if curl --fail --silent -H "Authorization: token $COMPOSER_GITHUB_OAUTH_TOKEN" https://api.github.com/rate_limit > /dev/null; then
+            ${PHP_DIR}/bin/php \
+              -d suhosin.executor.include.whitelist=phar \
+              -d suhosin.executor.func.blacklist=none \
+              -d disable_functions= \
+              /usr/local/bin/composer config -g github-oauth.github.com "$COMPOSER_GITHUB_OAUTH_TOKEN" &> /dev/null # redirect outdated version warnings (Composer sends those to STDOUT instead of STDERR)
+            echo 'Using $COMPOSER_GITHUB_OAUTH_TOKEN for GitHub OAuth.'
+        else
+            echo 'Invalid $COMPOSER_GITHUB_OAUTH_TOKEN for GitHub OAuth!'
+        fi
+    else
+        # Don't forget to remove any stored key if it's gone from the env
+        ${PHP_DIR}/bin/php \
+          -d suhosin.executor.include.whitelist=phar \
+          -d suhosin.executor.func.blacklist=none \
+          -d disable_functions= \
+          /usr/local/bin/composer config -g --unset github-oauth.github.com &> /dev/null # redirect outdated version warnings (Composer sends those to STDOUT instead of STDERR)
```

Comment:
why are removing stored keys? Won't this be a new container, and so nothing will be set here?